packages/react-reconciler/src/ReactFiberCompleteWork.new.js

```diff
@@ -1191,6 +1191,15 @@ function completeWork(
         }
       }
 
+      // If the suspended state of the boundary changes, we need to schedule
+      // a passive effect, which is when we process the transitions
+      if (nextDidTimeout !== prevDidTimeout) {
```

Comment:
Nit: The branch right below can be nested inside of this one so that in the case where there was no change, there are fewer checks.

```js
if (nextDidTimeout !== prevDidTimeout) {
  // Set Passive flag
  if (nextDidTimeout) {
    // Set Visibility flag
  }
}
```